app/assets/javascripts/host_edit.js

```diff
@@ -77,6 +75,22 @@ function computeResourceSelected(item){
   }
 }
 
+function handle_nic_updates() {
+  var modal_window = $('#interfaceModal');
+
+  var handler = function(event = null) {
+    update_nics(function() {
+      interface_subnet_selected(primary_nic_form().find('select.interface_subnet'));
+    });
+  };
+
+  if (modal_window.is(":visible")) {
+    modal_window.modal('hide').on('hidden.bs.modal', handler);
```

Comment:
I think this registers `handler` call multiple times, once after each compute resource update. `update_nics` seems to get called multiple_times then.